templates/tftmpl/tmplfunc/services_regex.go

```diff
@@ -185,6 +186,19 @@ func (d *servicesRegexQuery) Fetch(clients dep.Clients) (interface{}, *dep.Respo
 		opts.NodeMeta = d.nodeMeta
 	}
 
+	// This Fetch depends on multiple API calls for update. This adds an
+	// additional delay to process new updates to this query result. 1 second is
+	// used to account for Consul cluster propagation of the change at scale.
+	// https://www.hashicorp.com/blog/hashicorp-consul-global-scale-benchmark
+	//
+	// This affects template functions that have propagation depencies on
+	// services. KV query is not affected by this because it is a different
+	// entity within Consul.
+	//
+	// Without this delay, CatalogServices may have services that are not yet
+	// propagated to HealthServices
```

Comment:
Or the service has not returned healthy yet!

The [initial health status for checks is immediately set to "critical"](https://www.consul.io/docs/discovery/checks#initial-health-check-status) by default and has a delay to change healthy based on the checks.

Reply:
Oh yes! Thank you. I updated the comment to include the health status and the link you shared.